src/components/modal/index.tsx

```diff
@@ -32,6 +35,21 @@ export interface ModalProps {
 }
 
 export const Modal: React.FC<ModalProps> = ({ title, show, onExit, children }) => {
+    // Close modal on Esc press
+    const handleEscKey = useCallback((event: KeyboardEvent) => {
+        if (event.key === KEY_NAME_ESC) {
+            onExit();
+        }
+    }, [onExit]);
+
+    useEffect(() => {
+        document.addEventListener(KEY_EVENT_TYPE, handleEscKey, false);
+
+        return () => {
+            document.removeEventListener(KEY_EVENT_TYPE, handleEscKey, false);
+        };
+    }, [handleEscKey]);
```

Comment:
False is the default value, and I'd argue explicitly providing it doesn't have an impact on th legibility of the code.
```suggestion
    useEffect(() => {
        document.addEventListener(KEY_EVENT_TYPE, handleEscKey);

        return () => {
            document.removeEventListener(KEY_EVENT_TYPE, handleEscKey);
        };
    }, [handleEscKey]);
```